### PR TITLE
Added password checker

### LINKED ISSUE #21: Validation and strength checker for password
During registration add a password strength checker and add validation.
For example,
The password should have minimum 8 characters. Display appropriate message to the right of the textbox alerting the user
A password strength checker something like this:

ajax-password-strength-meter-script

### PR DESCRIPTION
**Pull request by rahulbisht12**
## Fixes 
#21 

## Description
Password strength (out of 4)

- Length at least 6 (mandatory)
- one uppercase letter
- one number
- one special character

Gives one point for each condition to be true

It gives the user suggestions also to make it strong also the password checker is only shown when the user is typing the password

Also I have fixed the indentation on the files in which I have made the changes. I have made it to 4 tabs.

## Reviewers
@AkankshaBodhankar

## Screenshots

p15
p16
p17
p18
p19
p20
p21
p22

### CODE REVIEW

**Comment by AkankshaBodhankar** [comment]
Can you post a screenshot

**Comment by rahulbisht12** [comment]
@AkankshaBodhankar Screenshots added :)

**Comment by AkankshaBodhankar** [comment]
Can you add a border to the meter. What all validations have you done?

File: registration.php [hunk 1]
```diff
@@ -45,25 +47,26 @@
...
+      header("location: welcome.php");
    }
-   
+
    require 'dbconnect.php';
-   
+
    if(isset($_POST['email'])&&isset($_POST['uname'])&&isset($_POST['password'])&&isset($_POST['host_country']))
    {
-
```
Comment (AkankshaBodhankar) [inline registration.php]: why is there extra line here

File: registration.php [hunk 2]
```diff
@@ -24,9 +24,11 @@
          <td><input class="input-box" type="text" id="uname" name="uname" placeholder="xyz123" required/></td>
        </tr>
         <tr>
-         <th class="text">Password:</th>
-         <td><input class="input-box" type="password" id="password" name="password" required/></td>
-       </tr>
+         <th class="text" style="vertical-align: top">Password:</th>
+         <td><input class="input-box" type="password" id="password" name="password" required/>
+           <meter max="4" id="password-strength-meter"></meter>
```
Comment (AkankshaBodhankar) [inline registration.php]: start <input> tag below the <td> tag

File: css files/loginAndRegistration.css [hunk 3]
```diff
@@ -136,3 +136,29 @@ a:hover {
...
+}
+
+meter::-webkit-meter-bar {
+    background: rgba(0, 0, 0, 0.1) none;
+}
+
+meter[value="1"]::-webkit-meter-optimum-value { background: red; }
+meter[value="2"]::-webkit-meter-optimum-value { background: yellow; }
+meter[value="3"]::-webkit-meter-optimum-value { background: orange; }
+meter[value="4"]::-webkit-meter-optimum-value { background: green; }
```
Comment (AkankshaBodhankar) [inline css files/loginAndRegistration.css]: change the green to the hex code  (Use the dark accent and post a screenshot)
Main green color: 05D197
Dark accent: 2DB58F
Darker accent: 02845F

**Comment by rahulbisht12** [comment]
@AkankshaBodhankar  I have made the required changes :)

p7

**Comment by AkankshaBodhankar** [comment]
Can you make the green a little dark. I still don't get a good feel about the color.

**Comment by rahulbisht12** [comment]
@AkankshaBodhankar I have made it to "Darker accent"

p8

File: javascripts/passwordStrengthChecker.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        text.innerText = "Too Strong"
+    }
+    else if (score == 3) {
+        text.innerText = "Strong";
+    }
+    else if (score == 2) {
+        text.innerText = "Weak";
+    }
+    else {
+        text.innerText = "Too Weak";
```
Comment (Anjali05) [inline javascripts/passwordStrengthChecker.js]: Change the text to "Very Weak"

**Comment by rahulbisht12** [comment]
@Anjali05 I have updated the password strength checker message. :)

File: css files/loginAndRegistration.css [hunk 5]
```diff
@@ -136,3 +136,40 @@ a:hover {
...
+meter[value="2"]::-webkit-meter-optimum-value { background: yellow; }
+meter[value="3"]::-webkit-meter-optimum-value { background: orange; }
+meter[value="4"]::-webkit-meter-optimum-value { background: #02845F; }
+
+meter[value="1"]::-moz-meter-bar { background: red; }
+meter[value="2"]::-moz-meter-bar { background: yellow; }
+meter[value="3"]::-moz-meter-bar { background: orange; }
+meter[value="4"]::-moz-meter-bar { background: #02845F; }
+
+#password-strength-text{
```
Comment (AkankshaBodhankar) [inline css files/loginAndRegistration.css]: Space before {

File: css files/loginAndRegistration.css [hunk 6]
```diff
@@ -136,3 +136,40 @@ a:hover {
...
+    font-size: 14px;
+    margin: 2px;
+}
+
+#suggestion{
```
Comment (AkankshaBodhankar) [inline css files/loginAndRegistration.css]: Spaces before {

**Comment by rahulbisht12** [comment]
@AkankshaBodhankar I have made the changes :)

**Comment by AkankshaBodhankar** [comment]
Looks good!

File: css files/loginAndRegistration.css [hunk 7]
```diff
@@ -136,3 +136,62 @@ a:hover {
     border-radius: 10px;
     -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, .3);
 }
+
```
Comment (AkankshaBodhankar) [inline css files/loginAndRegistration.css]: Add the tags according to alphabetical order not at the end of the file

Comment (rahulbisht12) [inline css files/loginAndRegistration.css]: @AkankshaBodhankar Ok, I will do it tomorrow morning and update the PR.

**Comment by rahulbisht12** [comment]
@AkankshaBodhankar I have updated the PR

**Comment by AkankshaBodhankar** [comment]
Okay

**Pull request merged**